src/ap/java/io/github/noeppi_noeppi/libx/annotation/processor/modinit/data/DatagenProcessor.java

```diff
@@ -17,6 +17,10 @@ public static void processDatagen(Element element, ModEnv env) {
             env.messager().printMessage(Diagnostic.Kind.ERROR, "Can't use @Datagen on element that is not a class.", element);
             return;
         }
+        if (element.getModifiers().contains(Modifier.PUBLIC)) {
```

Comment:
Not sure about how it’s actually working but shouldn’t this condition be inverted? Otherwise it checks for "is it public?", if yes it complains that it’s not. But not sure about that, correct me if I’m wrong.